lib/web3/filter.js

```diff
@@ -83,51 +84,69 @@ var Filter = function (options, methods, formatter) {
     this.implementation = implementation;
     this.callbacks = [];
     this.formatter = formatter;
-    this.filterId = this.implementation.newFilter(this.options);
+    this.implementation.newFilter(this.options, function(error, id){
+        if(error)
+            self.filterError = error;
+        else
+            self.filterId = id;
+    });
 };
 
 Filter.prototype.watch = function (callback) {
-    this.callbacks.push(callback);
     var self = this;
 
-    var onMessage = function (error, messages) {
-        if (error) {
-            return self.callbacks.forEach(function (callback) {
-                callback(error);
-            });
-        }
+    // check inf an interval of 10ms if the filter id has arrived
+    var intervalId = setInterval(function(){
```

Comment:
setInterval as setTimeout was implement in otto, otherwise out filters wouldn't work anyway.